sktime/transformations/series/tsfel/_tsfel.py

```diff
@@ -0,0 +1,307 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""TSFEL transformer to extract features by domain or specific feature names."""
+
+__author__ = ["Faakhir30"]
+
+
+import pandas as pd
+
+from sktime.transformations.base import BaseTransformer
+
+
+class TSFELTransformer(BaseTransformer):
+    """TSFEL transformer to extract features by domain or specific feature names.
+
+    This transformer uses `features` parameter to extract features in following ways:
+
+    1. By domain: Pass a domain ('statistical', 'temporal', 'spectral', 'fractal')
+    2. By feature names: Pass feature function names (e.g., 'abs_energy', 'auc')
+    3. Mixed: Pass a list containing both domain strings and feature names
+
+    For domain-based extraction, uses TSFEL's `time_series_features_extractor`.
+    For individual features, calls the feature functions directly from
+    `tsfel.feature_extraction.features`.
+
+    See tsfel documentation for available options for features and parameters.
+    https://tsfel.readthedocs.io/en/latest/descriptions/feature_list.html
+
+    Parameters
+    ----------
+    features : str, list of str, or None, optional (default=None)
+        Features to extract. Can be:
+
+        - A domain string: 'statistical', 'temporal', 'spectral', 'fractal'
+        - A list of feature function names: ['abs_energy', 'auc', 'autocorr']
+        - A list mixing domains and features: ['statistical', 'abs_energy']
+        - None: extract all features from all domains
+
+    kwargs : dict, optional (default=None)
+        Additional keyword arguments passed to tsfel's feature extractor or
+        individual feature functions. Common parameters include:
+
+        - fs : float, sampling frequency
+        - window_size : int, size of windows for feature extraction
+        - overlap : float, overlap between windows (0-1)
+        - verbose : int, verbosity level (0 or 1)
+        - Any feature-specific parameters (e.g., percentile for ecdf_percentile_count)
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.tsfel import TSFELTransformer
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> # Extract all statistical domain features
+    >>> transformer = TSFELTransformer(
+    ...     features="statistical", kwargs={"verbose": 0}
+    ... )
+    >>> features = transformer.fit_transform(y)  # doctest: +SKIP
+    >>> # Access TSFEL output for feature
+    >>> transformer['statistical'].iloc[0]  # doctest: +SKIP
+    >>> # Extract feature with custom parameters
+    >>> transformer = TSFELTransformer(
+    ...     features=["ecdf_percentile_count"],
+    ...     kwargs={"percentile": [0.6, 0.9, 1.0], "verbose": 0}
+    ... )
+    >>> features = transformer.fit_transform(y)  # doctest: +SKIP
+    >>> transformer['ecdf_percentile_count'].iloc[0]  # doctest: +SKIP
+    """
+
+    _tags = {
+        "authors": ["Faakhir30"],
```

Comment:
please also add the `tsfel` authors for credit - you can put yourself as `maintainer` if you want